src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -283,16 +283,20 @@ void Run (DirectoryAssemblyResolver res, bool useMarshalMethods)
 					TypeDefinition conflict;
 					bool hasConflict = false;
 					if (managed.TryGetValue (managedKey, out conflict)) {
-						if (!managedConflicts.TryGetValue (managedKey, out var list))
-							managedConflicts.Add (managedKey, list = new List<string> { conflict.GetPartialAssemblyName (cache) });
-						list.Add (type.GetPartialAssemblyName (cache));
+						if (!conflict.Module.Mvid.Equals(mvid)) {
```

Comment:
Does this work here:
```suggestion
						if (!conflict.Module.Mvid.Equals(type.Module.Mvid)) {
```